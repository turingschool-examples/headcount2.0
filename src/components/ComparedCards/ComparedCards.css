.compared-cards-container {
  position: fixed;
  padding-top: 25px;
  border-top-right-radius: 15px;
  border-bottom-right-radius: 15px;
  background-color: rgba(5, 5, 5, 1);
  border-color: rgba(5, 5, 5, 1);
  bottom: 10%;
  width: 825px;
}

.compared-cards {
  display: flex;
  flex-direction: row;
  align-content: space-between;
  width: 100%;
}

.compared-cards article {
  margin: 10px;
}

#remove-comparison {
  border: 3px solid #D23D54;
  background: none;
  color: #D23D54;
  margin-bottom: 10px;
  padding: 3px 3px 5px 3px;
}

.clicked {
  background-color: rgba(5, 5, 5, 1);
}

@media only screen and (max-width: 825px) {
  .compared-cards-container {
    position: relative;
    width: 280px;
    margin: 10px auto;
    border-radius: 15px;
  }

  .compared-cards article {
    margin: 10px auto;
  }

  .compared-cards {
    flex-direction: column;
  }
}