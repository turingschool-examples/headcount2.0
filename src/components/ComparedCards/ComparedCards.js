import React from 'react';
import Card from '../Card/Card.js';
import './ComparedCards.css';
import ComparisonCard from '../ComparisonCard/ComparisonCard';

const ComparedCards = ({ selectedCards, selectCard, makeComparison, removeComparison }) => {
  
  let comparedObject;
  if (selectedCards.length > 1) {
    comparedObject = makeComparison(selectedCards[0].location, selectedCards[1].location)
  }
  const selectedCardsArray = selectedCards.map((district, key) => 
    <Card 
      districtName={district.location}
      data={district.data}
      id={district.location}
      selectCard={selectCard}
      key={key}
    />);

  return (

    <article className="compared-cards-container">
      <div className="compared-cards">
        {selectedCardsArray[0]}
        <ComparisonCard comparedObject={comparedObject}/>
        {selectedCardsArray[1]}
      </div>
      <div>
        <button 
          id="remove-comparison"
          onClick={removeComparison}>
            clear comparison
        </button>
      </div>
    </article>
  )
}

export default ComparedCards;