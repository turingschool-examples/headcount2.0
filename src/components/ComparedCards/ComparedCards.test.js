/* eslint-disable */
import React from 'react';
import ComparedCards from './ComparedCards';
import { shallow } from 'enzyme';

describe('ComparedCards', () => {

  it('snapshot test', () => {
    const mockData = [{data: {2004: .523, 2014: .321}, location: 'COLORADO'}, {data: {2004: .283, 2014: .432}, location: 'ACADEMY 20'}];
    const mockFunction = jest.fn();
    const wrapper = shallow(<ComparedCards
                              selectCard={mockFunction}
                              removeComparison={mockFunction} 
                              selectedCards={mockData}
                              makeComparison={mockFunction}/>);
    expect(wrapper).toMatchSnapshot();
  })

  it('should render two card components if there is two cards in the data', () => {
    const mockData = [{data: {2004: .523, 2014: .321}, location: 'COLORADO'}, {data: {2004: .283, 2014: .432}, location: 'ACADEMY 20'}];
    const mockFunction = jest.fn();

    const wrapper = shallow(<ComparedCards
                              selectCard={mockFunction}
                              removeComparison={mockFunction} 
                              selectedCards={mockData}
                              makeComparison={mockFunction}/>);
    expect(wrapper.find('Card').length).toEqual(2);
  });

});