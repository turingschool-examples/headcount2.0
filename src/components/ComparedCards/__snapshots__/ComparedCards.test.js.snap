// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComparedCards snapshot test 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ComparedCards
    makeComparison={[Function]}
    removeComparison={[Function]}
    selectCard={[Function]}
    selectedCards={
        Array [
            Object {
              "data": Object {
                "2004": 0.523,
                "2014": 0.321,
              },
              "location": "COLORADO",
            },
            Object {
              "data": Object {
                "2004": 0.283,
                "2014": 0.432,
              },
              "location": "ACADEMY 20",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="compared-cards"
>
          <Card
                    data={
                              Object {
                                        "2004": 0.523,
                                        "2014": 0.321,
                                      }
                    }
                    districtName="COLORADO"
                    id="COLORADO"
                    selectCard={[Function]}
          />
          <ComparisonCard
                    comparedObject={undefined}
          />
          <Card
                    data={
                              Object {
                                        "2004": 0.283,
                                        "2014": 0.432,
                                      }
                    }
                    districtName="ACADEMY 20"
                    id="ACADEMY 20"
                    selectCard={[Function]}
          />
</div>,
        <div>
          <button
                    id="remove-comparison"
                    onClick={[Function]}
          >
                    clear comparison
          </button>
</div>,
      ],
      "className": "compared-cards-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Card
              data={
                            Object {
                                          "2004": 0.523,
                                          "2014": 0.321,
                                        }
              }
              districtName="COLORADO"
              id="COLORADO"
              selectCard={[Function]}
/>,
            <ComparisonCard
              comparedObject={undefined}
/>,
            <Card
              data={
                            Object {
                                          "2004": 0.283,
                                          "2014": 0.432,
                                        }
              }
              districtName="ACADEMY 20"
              id="ACADEMY 20"
              selectCard={[Function]}
/>,
          ],
          "className": "compared-cards",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "function",
            "props": Object {
              "data": Object {
                "2004": 0.523,
                "2014": 0.321,
              },
              "districtName": "COLORADO",
              "id": "COLORADO",
              "selectCard": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "comparedObject": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "function",
            "props": Object {
              "data": Object {
                "2004": 0.283,
                "2014": 0.432,
              },
              "districtName": "ACADEMY 20",
              "id": "ACADEMY 20",
              "selectCard": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <button
            id="remove-comparison"
            onClick={[Function]}
>
            clear comparison
</button>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "clear comparison",
            "id": "remove-comparison",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "clear comparison",
          "type": "button",
        },
        "type": "div",
      },
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="compared-cards"
>
            <Card
                        data={
                                    Object {
                                                "2004": 0.523,
                                                "2014": 0.321,
                                              }
                        }
                        districtName="COLORADO"
                        id="COLORADO"
                        selectCard={[Function]}
            />
            <ComparisonCard
                        comparedObject={undefined}
            />
            <Card
                        data={
                                    Object {
                                                "2004": 0.283,
                                                "2014": 0.432,
                                              }
                        }
                        districtName="ACADEMY 20"
                        id="ACADEMY 20"
                        selectCard={[Function]}
            />
</div>,
          <div>
            <button
                        id="remove-comparison"
                        onClick={[Function]}
            >
                        clear comparison
            </button>
</div>,
        ],
        "className": "compared-cards-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Card
                data={
                                Object {
                                                "2004": 0.523,
                                                "2014": 0.321,
                                              }
                }
                districtName="COLORADO"
                id="COLORADO"
                selectCard={[Function]}
/>,
              <ComparisonCard
                comparedObject={undefined}
/>,
              <Card
                data={
                                Object {
                                                "2004": 0.283,
                                                "2014": 0.432,
                                              }
                }
                districtName="ACADEMY 20"
                id="ACADEMY 20"
                selectCard={[Function]}
/>,
            ],
            "className": "compared-cards",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "0",
              "nodeType": "function",
              "props": Object {
                "data": Object {
                  "2004": 0.523,
                  "2014": 0.321,
                },
                "districtName": "COLORADO",
                "id": "COLORADO",
                "selectCard": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "comparedObject": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "function",
              "props": Object {
                "data": Object {
                  "2004": 0.283,
                  "2014": 0.432,
                },
                "districtName": "ACADEMY 20",
                "id": "ACADEMY 20",
                "selectCard": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <button
              id="remove-comparison"
              onClick={[Function]}
>
              clear comparison
</button>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "clear comparison",
              "id": "remove-comparison",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "clear comparison",
            "type": "button",
          },
          "type": "div",
        },
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
